### PR TITLE
`Discussion`: Add discussion related features to Artemis user documentation

### PR DESCRIPTION
**Pull request by lschlesinger**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

We need to add the discussion related feature to the user manual in order to make users aware of them and explain how to make use of them.

### Description

I added a new section to the documentation including all the relevant information for the user as well as explanatory screenshots.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Check the diff for inconsistencies, bad language, typos etc.
2. Check the resulting, updated documentation to see how the changes would _look like_. You can simply click the according job in the github pipeline to be forwarded to the documentation created out of the current PR version:
Bildschirmfoto 2021-12-01 um 23 52 30

#### Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

File: docs/user/discussion.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+This can be done by clicking the checkmark next to the answer post.
+Note, that only the author of the post as well as a moderator can perform this action.
+This is helpful for moderators to identify open questions, e.g., by applying the according filter in the course overview.
+It also highlights the correct answer for other students that have a similar problem and search for a suitable solution.
+
+Tag Your Post
+^^^^^^^^^^^^^
+
+When creating a post, users can choose to add arbitrary tags.
+Tagging a post will further narrow down the post purpose or content in precise and descriptive buzzwords, that might follow a course-specific taxonomy.
```
Comment (b-fein) [inline docs/user/discussion.rst:104]:
```suggestion
Tagging a post will further narrow down the post purpose or content in precise and descriptive keywords, that might follow a course-specific taxonomy.
```
‘Buzzword’ has the fairly negative connotation of being overused. I don’t think this meaning is intended here.

**Review by b-fein — APPROVED** [review]
Looks good. I added a few inline comments with only small suggestions.

File: docs/user/discussion.rst (hunk 1 above)
Comment (lschlesinger) [reply docs/user/discussion.rst]: changed, thanks!

File: docs/user/discussion.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+To foster interaction we integrate the well-known emoji reaction bar.
+Each user in the course can react on every post by making use of the emoji selection button.
+The ``+`` emoji serves as up-voting reaction, which influences the display order.
+
+Reference Posts
+^^^^^^^^^^^^^^^
+
+If users want to refer to other posts, they can integrate a simple pattern including the hashtag (``#``) combined with the post identifier.
+A post's identifier is appended to the post title (such as seen in the screenshot).
```
[the remark this answers is not in the document]
Comment (lschlesinger) [reply docs/user/discussion.rst]: changed!

File: docs/user/discussion.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+Course-wide Posts & Course Discussion Overview
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Besides lecture or exercise related questions, Artemis offers a third type of posts: posts with course-wide topics, such as ``Organization`` or ``Tech Support``.
+These posts can only be created on the course discussion overview which is showed in the screenshot below.
+
+The ``Discussion`` space of an Artemis course serves as overview for *all* posts in a course.
+Hence, course-wide posts as well as exercise posts and lecture posts are listed.
+Here, users can easily query, sort and filter existing posts.
```
[the remark this answers is not in the document]
Comment (lschlesinger) [reply docs/user/discussion.rst]: 👍

File: docs/user/discussion.rst [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+Exemplary exercise detail page with posts:
+
+|exercise-posts|
+
+Course-wide Posts & Course Discussion Overview
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Besides lecture or exercise related questions, Artemis offers a third type of posts: posts with course-wide topics, such as ``Organization`` or ``Tech Support``.
+These posts can only be created on the course discussion overview which is showed in the screenshot below.
```
[the remark this answers is not in the document]
Comment (lschlesinger) [reply docs/user/discussion.rst]: thanks

File: docs/user/discussion.rst [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    :local:
+    :depth: 2
+
+Overview
+--------
+
+Artemis enables students, tutors, and instructors to actively participate in discussions.
+Various discussion features allow students to engage with peers and ask all kinds of questions whereas moderators (e.g., instructors and tutors) can provide general course information and answer content-related questions.
+Discussions can be started in different contexts, namely for lectures, exercises, or courses.
+Below, you find more information on specific features and how to use them.
```
[the remark this answers is not in the document]
Comment (lschlesinger) [reply docs/user/discussion.rst]: changed!

**Comment by lschlesinger** [comment]
@b-fein thanks for the helpful comments, integrated all of them ;)!!

**Review by b-fein — APPROVED** [review]

File: docs/user/discussion.rst [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On the course discussion overview, a user can query *all* existing posts by different criteria.
+This can be a text-based search, context filters (i.e., a certain lecture, exercise or course-wide topic), or other post characteristics such as if it is already resolved.
+By using different sort configurations, users can find the information they are looking for at the top of the list: for example the post that got the most votes, or the post that was created most recently.
+
+React on Posts
+^^^^^^^^^^^^^^
+
+To foster interaction we integrate the well-known emoji reaction bar.
```
Comment (Strohgelaender) [inline docs/user/discussion.rst:73]:
```suggestion
To encourage interaction we provide the well-known emoji reaction bar.
```

**Review by Strohgelaender — APPROVED** [review]
Changes look good, I only added one small sugesstion:

**Review by VinnavanSembian — APPROVED** [review]
Everything looks good to me!

**Review by lBruchmann — APPROVED** [review]
Everything looks good to me. Great user documentation!

**Review by mallgaier — APPROVED** [review]
Reviewed together with @Strohgelaender , @lBruchmann , @VinnavanSembian and @willergeorg . Very good and detailed user documentation!

**Review by willergeorg — APPROVED** [review]
Thanks for extending the Docs here! Changes look good to me.

**Pull request merged**